### PR TITLE
Add note on task message validation

### PR DESCRIPTION
**Pull request by MetRonnie**
Sister PR to https://github.com/cylc/cylc-flow/pull/3820

### CODE REVIEW

File: src/user-guide/task-implementation.rst [hunk 1]
```diff
@@ -95,23 +95,22 @@ the ``$PATH`` in your bash configuration files (e.g. ``.bashrc``).
 Task Messages
 -------------
 
-Task jobs send status messages back to the server program to report that
+Task jobs send status messages back to the :term:`scheduler` to report that
```
Comment (MetRonnie) [inline src/user-guide/task-implementation.rst:98]: Am I correct that server program should be renamed to scheduler?

File: src/user-guide/task-implementation.rst [hunk 2]
```diff
@@ -95,23 +95,22 @@ the ``$PATH`` in your bash configuration files (e.g. ``.bashrc``).
...
 execution has started, succeeded, or failed. Custom messages can also be sent
 by the same mechanism, with various severity levels. These can be used to
-trigger other tasks off specific task outputs, or to trigger execution of
-event handlers by the server program (see :ref:`EventHandling`), or just to
-write information to the server log.
+trigger other tasks off specific task outputs(see (:ref:`MessageTriggers`), or
```
Comment (oliver-sanders) [inline src/user-guide/task-implementation.rst:101]:
```suggestion
trigger other tasks off specific task outputs (see :ref:`MessageTriggers`), or
```

File: src/user-guide/task-implementation.rst (hunk 1 above)
Comment (oliver-sanders) [reply src/user-guide/task-implementation.rst:98]: You are indeed!

**Review by hjoliver — APPROVED** [review]
Approving this as it is only loosely dependent on the as yet undecided message validation question https://github.com/cylc/cylc-flow/pull/3820#discussion_r489598087

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**